File: src/main/java/com/ambition/business/content/controller/ContentArticleController.java
package com.ambition.business.content.controller;

import com.ambition.business.content.domain.ContentArticle;
import com.ambition.business.content.service.IContentArticleService;
import com.ambition.business.user.domain.SysUser;
import com.ambition.common.annotations.AddSysLog;
import com.ambition.common.annotations.CurrentUser;
import com.ambition.common.annotations.LoginedUser;
import com.ambition.common.util.R;
import io.swagger.annotations.ApiOperation;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import springfox.documentation.annotations.ApiIgnore;

import javax.validation.Valid;
import java.util.List;

/**
 * <p>
 *  前端控制器
 * </p>
 *
 * @author wuys
 * @since 2019-12-11
 */
@RestController
@RequestMapping("/content-article")
public class ContentArticleController {

     private static final Logger LOG = LoggerFactory.getLogger(ContentArticleController.class);

     @Autowired
     private IContentArticleService service;

    /**
    * 查询分页数据
    */
    @ApiOperation(value = "查询分页数据")
    @GetMapping(value = "/list")
    @LoginedUser
    @AddSysLog(descrption = "分页查询ContentArticle列表")
    public R findListByPage(@CurrentUser@ApiIgnore SysUser sysUser,@RequestParam(name = "page", defaultValue = "1") int page,@RequestParam(name = "pageSize", defaultValue = "20") int pageSize){
      return service.findListByPage(page,pageSize,sysUser.getGroupId());
    }


     /**
     * 根据id查询
     */
     @ApiOperation(value = "根据id查询数据")
     @GetMapping(value = "/getById")
     @LoginedUser
     @AddSysLog(descrption = "根据id查询ContentArticle")
     public R getById(@RequestParam("pkid") Long pkid){
       return service.getById(pkid);
     }

     /**
     * 新增
     */
     @ApiOperation(value = "新增数据")
     @RequestMapping(value = "/add", method = RequestMethod.POST)
     @LoginedUser
     @AddSysLog(descrption = "新增ContentArticle")
     public R save(@CurrentUser@ApiIgnore SysUser sysUser,@RequestBody @Valid ContentArticle entity){
        return service.saveContentArticle(entity);
     }

     /**
     * 删除
     */
     @ApiOperation(value = "删除数据")
     @RequestMapping(value = "/del",method = RequestMethod.POST)
     @LoginedUser
     @AddSysLog(descrption = "根据id删除ContentArticle")
     public R delete(@RequestParam("ids") List<Long> ids){
       return service.deleteByIds(ids);
     }

     /**
     * 修改
     */
     @ApiOperation(value = "更新数据")
     @RequestMapping(value = "/update", method = RequestMethod.POST)
     @LoginedUser
     @AddSysLog(descrption = "更新ContentArticle")
     public R update(@CurrentUser@ApiIgnore SysUser sysUser,@RequestBody @Valid ContentArticle entity){
       return service.updateContentArticleById(entity);
     }
}

File: src/main/java/com/ambition/business/content/controller/ContentArticleController.kt
package com.ambition.business.content.controller

import com.ambition.business.content.service.IContentArticleService
import com.ambition.business.user.domain.SysUser
import com.ambition.common.annotations.AddSysLog
import com.ambition.common.annotations.CurrentUser
import com.ambition.common.annotations.LoginedUser
import com.ambition.common.util.R
import io.swagger.annotations.ApiOperation
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import springfox.documentation.annotations.ApiIgnore

/**
 * <pre>
 *
 *
 * @title: ContentArticleController
 * @description:
 * @author: wuys
 * @datetime: 2019-12-18 9:41
 * </pre>
 */
@RestController
@RequestMapping("/{origin}/content")
class ContentArticleController(
        @Autowired private val service: IContentArticleService
) {

    private companion object{
        val LOG = LoggerFactory.getLogger(ContentArticleController::class.java)
    }

    @ApiOperation(value = "查询分页数据")
    @GetMapping("/list")
    @LoginedUser
    @AddSysLog(descrption = "分页查询ContentArticle列表")
    fun findListByPage(@CurrentUser @ApiIgnore sysUser:SysUser, @RequestParam(name = "page", defaultValue = "1") page:Int, @RequestParam(name = "pageSize", defaultValue = "20") pageSize:Int):R {
        LOG.info("find list by page!")
        return service.findListByPage(page,pageSize,sysUser.groupId)
    }

}
